### PR TITLE
Add createTypeAliasDeclaration and updateTypeAliasDeclaration factories

### LINKED ISSUE #15489: add ts.createTypeAliasDeclaration
**TypeScript Version:**  2.3.2
Code

```ts
type Foo = {}
```

How can I generate a node that is equivalent of above source code?

### PR DESCRIPTION
**Pull request by mohsen1**
Fixes #15489

### CODE REVIEW

File: src/compiler/factory.ts [hunk 1]
```diff
@@ -1462,6 +1462,22 @@ namespace ts {
...
 
+    export function createTypeAliasDeclaration(name: string | Identifier, typeParameters: TypeParameterDeclaration[] | undefined, type: ts.TypeNode) {
+        const node = <TypeAliasDeclaration>createSynthesizedNode(SyntaxKind.TypeAliasDeclaration);
+        node.name = asName(name);
+        node.typeParameters = asNodeArray(typeParameters);
+        node.type = type;
+        return node;
+    }
+
+    export function updateTypeAliasDeclaration(node: TypeAliasDeclaration, name: string | Identifier, typeParameters: TypeParameterDeclaration[] | undefined, type: ts.TypeNode) {
```
Comment (rbuckton) [inline src/compiler/factory.ts]: `name` should just be `Identifier` here. Also, you do not need to qualify `TypeNode` with the `ts.` namespace.

**Comment by rbuckton** [comment]
The PR builds failed. Did you run all of the tests?

**Comment by mohsen1** [comment]
Failures don't seem to be related.

File: src/compiler/factory.ts [hunk 2]
```diff
@@ -1462,15 +1462,15 @@ namespace ts {
             : node;
     }
 
-    export function createTypeAliasDeclaration(name: string | Identifier, typeParameters: TypeParameterDeclaration[] | undefined, type: ts.TypeNode) {
+    export function createTypeAliasDeclaration(name: Identifier, typeParameters: TypeParameterDeclaration[] | undefined, type: TypeNode) {
```
Comment (phated) [inline src/compiler/factory.ts]: This should still support `string`.  The update method is the only one that only accepts `Identifier`

Comment (mohsen1) [inline src/compiler/factory.ts]: Updated. Thanks!

**Comment by mohsen1** [comment]
@rbuckton is this g2g?

**Comment by rbuckton** [comment]
Looks good!

**Pull request merged**